src/tools/genpolicy/src/yaml.rs

```diff
@@ -264,10 +264,23 @@ pub fn add_policy_annotation(
     let policy_key = serde_yaml::Value::String("io.katacontainers.config.agent.policy".to_string());
     let policy_value = serde_yaml::Value::String(policy.to_string());
 
-    let path_components = metadata_path.split('.');
-    for name in path_components {
-        ancestor = ancestor.get_mut(name).unwrap();
+    if !metadata_path.is_empty() {
```

Comment:
It seems to not make sense `metadata_path` being `empty`, if so, then it is an error?

Reply:
It's empty for the caller from pod.rs.

Reply:
sorry, missed that. Yeah, for pod spec you need to include `metadata` in the yaml's root. ok.